### ISSUE TITLE
[CI] DiscoveryNodeTests.testDiscoveryNodeRoleWithOldVersion failure

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#64385]
**Build scan**: https://gradle-enterprise.elastic.co/s/25v4ftusogklk

**Repro line**:
> ./gradlew ':server:test' --tests "org.elasticsearch.cluster.node.DiscoveryNodeTests.testDiscoveryNodeRoleWithOldVersion" -Dtests.seed=DF4A3BD1A29F3AD3 -Dtests.security.manager=true -Dtests.locale=ar-IQ -Dtests.timezone=SystemV/CST6CDT -Druntime.java=8

**Reproduces locally?**:
Yes

**Applicable branches**:
7.10

**Failure history**:

This just started failing because of commit 1126ba4df8658fe9afef07629b103f5e766db5df
**Failure excerpt**:
```
org.elasticsearch.cluster.node.DiscoveryNodeTests > testDiscoveryNodeRoleWithOldVersion FAILED |  
-- | --
  | java.lang.IllegalStateException: unexpected byte [0xc3] |  
  | at __randomizedtesting.SeedInfo.seed([DF4A3BD1A29F3AD3:DFD7A5F1EC6E352A]:0) |  
  | at org.elasticsearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:575) |  
  | at org.elasticsearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:565) |  
  | at org.elasticsearch.cluster.node.DiscoveryNode.<init>(DiscoveryNode.java:281) |  
  | at org.elasticsearch.cluster.node.DiscoveryNodeTests.testDiscoveryNodeRoleWithOldVersion(DiscoveryNodeTests.java:136)
```

It fails because the test tries to deserialise a [v7.9 stream](https://github.com/elastic/elasticsearch/blob/1126ba4df8658fe9afef07629b103f5e766db5df/server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java#L131) using [v7.10 layout](https://github.com/elastic/elasticsearch/blob/1126ba4df8658fe9afef07629b103f5e766db5df/server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java#L135).

**Comment by ywangd** [comment]
Muted with 533b929e6c66f169bc5b96d2b165adb81a559e09

**Comment by jasontedor** [comment]
Closed by 6b119a43c1e9a0e93f758f12708e2705e3f358f5.
